compaction/compaction.cc

```diff
@@ -1751,6 +1752,10 @@ compact_sstables(sstables::compaction_descriptor descriptor, compaction_data& cd
         // Bypass the usual compaction machinery for dry-mode scrub
         return scrub_sstables_validate_mode(std::move(descriptor), cdata, table_s);
     }
+    if (descriptor.options.type() == compaction_type::Scrub
+            && std::get<compaction_type_options::scrub>(descriptor.options.options()).operation_mode == compaction_type_options::scrub::mode::abort) {
+                utils::get_local_injector().inject("rest_api_keyspace_scrub_abort", [&table_s] { throw compaction_aborted_exception(table_s.schema()->ks_name(), table_s.schema()->cf_name(), "scrub compaction found invalid data"); });
+    }
```

Comment:
This place is wrong for injecting the error since normally there is no reason to abort here.
Let's inject inline in the code itself where we check if scrub needs to abort (and the error injection should simulate a non-recoverable error for scrub).
Then it can apply also to other scrub modes, not only `scrub::mode::abort`.